keg_auth/model/__init__.py

```diff
@@ -461,6 +466,31 @@ def _query_permission_mapping(cls):
         return sa.union(direct, via_bundle)
 
 
+class AttemptMixin(object):
+    users_tablename = 'users'
+
+    @declared_attr
```

Comment:
We don't want to link directly to a user here. Rate-limiting mitigates against a certain vector of attacks, e.g. dictionary attacks on the username or password.

In the case of a user entering a username that doesn't exist, it appears we are a) not logging the attempt, and b) no rate-limiting is happening. What we really need to be doing is:
- use a string identifier field for what the user put in on the rate-limited form
- optionally have a user_id for ease of later reporting, but make it nullable
- log all attempts, whether the input was matched or not